pkg/client/cache/store.go

```diff
@@ -63,12 +66,38 @@ func MetaNamespaceKeyFunc(obj interface{}) (string, error) {
 	return meta.Name(), nil
 }
 
+// ReplaceIfNewer is a CanReplace implementation which requires both a and b to
+// implement meta.Interface. It returns true if ResourceVersion of a is greater than
+// ResourceVersion of b, and returns false otherwise.
+func ReplaceIfNewer(a, b interface{}) (bool, error) {
+	metaA, errA := meta.Accessor(a)
+	if errA != nil {
+		return false, fmt.Errorf("object has no meta: %v", errA)
+	}
+
+	metaB, errB := meta.Accessor(b)
+	if errB != nil {
+		return false, fmt.Errorf("object has no meta: %v", errB)
+	}
+
+	return metaA.ResourceVersion() > metaB.ResourceVersion(), nil
```

Comment:
ResourceVersion is a string, this almost certainly isn't doing what you want it to. Clients should treat ResourceVersion as an opaque blob, anyway, so I think this method of determining recency is not feasible.